components/organisms/ToolList/nav-item.tsx

```diff
@@ -21,7 +24,11 @@ const NavItem: React.FC<NavItemProps> = ({ username, filterName, tool, selectedF
   return (
     <Link
       href={`/${username ? `pages/${username}/` : ""}${filterName}/${tool.name.toLowerCase()}${
-        selectedFilter ? `/filter/${Array.isArray(selectedFilter) ? selectedFilter.join("/") : selectedFilter}` : ""
+        selectedFilter
+          ? `/filter/${Array.isArray(selectedFilter) ? selectedFilter.join("/") : selectedFilter}?range=${
+              range ?? "30"
+            }`
+          : `?range=${range ?? "30"}`
```

Comment:
refactor (non-blocking): This in conjunction with `const { range = "30" } = router.query` above.

```suggestion
          : `?range=${range}`
```